rohmu/notifier/http.py

```diff
@@ -0,0 +1,98 @@
+"""Copyright (c) 2022 Aiven, Helsinki, Finland. https://aiven.io/"""
+from .interface import Notifier
+from dataclasses import dataclass
+from datetime import datetime
+from enum import Enum, unique
+from queue import SimpleQueue
+from typing import Any, Dict, Optional
+
+import requests
+import threading
+
+
+@unique
+class Operation(Enum):
+    UPLOAD = "UPLOAD"
+    DELETE = "DELETE"
+    DELETE_TREE = "DELETE_TREE"
+
+
+def _get_requests_session() -> requests.Session:
+    retry = requests.adapters.Retry(
+        total=3,
+        backoff_factor=0.5,
+        status_forcelist=[429, 500, 502, 503, 504],
+        method_whitelist={"POST"},
+    )
+    adapter = requests.adapters.HTTPAdapter(max_retries=retry)
+    request_session = requests.Session()
+    request_session.mount("http://", adapter)
+    return request_session
+
+
+class HTTPNotifier(Notifier):
+    def __init__(self, url: str) -> None:
+        self._session = _get_requests_session()
+        self._url = url
+
+    def object_created(self, key: str, size: Optional[int]) -> None:
+        self._session.post(
+            self._url,
+            {
+                "op": Operation.UPLOAD.value,
+                "key": key,
+                "size": size,
+                "last_modified": datetime.now(),
+            },
+        )
+
+    def object_deleted(self, key: str) -> None:
+        self._session.post(self._url, {"op": Operation.DELETE.value, "key": key})
+
+    def tree_deleted(self, key: str) -> None:
+        self._session.post(self._url, {"op": Operation.DELETE_TREE.value, "key": key})
```

Comment:
Building on Sumit's comments would these also break rhome uploads if there is an error, as they're called inline in the main thread?